packages/vscode-extension/src/handlers.ts

```diff
@@ -795,6 +797,143 @@ export async function activateEnvironment(env: string): Promise<Result<Void, FxE
   return result;
 }
 
+export async function grantPermission(env: string): Promise<Result<Void, FxError>> {
+  let result: Result<any, FxError> = ok(Void);
+  ExtTelemetry.sendTelemetryEvent(TelemetryEvent.GrantPermission);
+
+  const eventName = ExtTelemetry.stageToEvent(Stage.grantPermission);
+  try {
+    const checkCoreRes = checkCoreNotEmpty();
+    if (checkCoreRes.isErr()) {
+      throw checkCoreRes.error;
+    }
+
+    const inputs: Inputs = getSystemInputs();
+    inputs.env = env;
+
+    result = await core.grantPermission(inputs);
+    if (result.isErr()) {
+      throw result.error;
+    }
+    window.showInformationMessage(
+      `Added account: '${inputs.email}'' to the environment '${env}' as a collaborator`
+    );
+
+    updateCollaboratorList(env);
+  } catch (e) {
+    result = wrapError(e);
+  }
+
+  await processResult(eventName, result);
```

Comment:
where do we send the success/error telemetry for grant?

Reply:
processResult function will help show error message dialog and send telemetry